### PR TITLE
Chore: Adapt to new AA and Nemo

### LINKED ISSUE #759: Adapt `residue_ring` dispatches introduced here
... to also return the projection map.
See https://github.com/Nemocas/AbstractAlgebra.jl/pull/1559 and https://github.com/Nemocas/Nemo.jl/pull/1636.

cc @thofma

### PR DESCRIPTION
**Pull request by lgoettgens**
Closes https://github.com/oscar-system/Singular.jl/pull/755, resolves #759

### CODE REVIEW

**Review by fingolfin — APPROVED** [review]

**Comment by lgoettgens** [comment]
This unfortunately needs changes to the tests that make it incompatible with Nemo 0.39

**Comment by thofma** [comment]
> This unfortunately needs changes to the tests that make it incompatible with Nemo 0.39

Depends on how dirty it should get. `R = tuple(residue_ring(Nemo.ZZ, 15)...)[1]` would work for both versions.

**Comment by lgoettgens** [comment]
Yes, there are workarounds for this. But @fingolfin and I weren't sure if this is worth the effort right now.
https://github.com/oscar-system/Singular.jl/issues/759 will be breaking anyway.

Maybe I can add a small workaround for the few cases changed here (to make it compatible with old and new Nemo for now) and once https://github.com/oscar-system/Singular.jl/issues/759 gets resolved in a breaking way, the compat get adapted and the workarounds removed.
Sounds like a plan?

**Comment by thofma** [comment]
I already have everything for #759 ready. Should I just push it here?

**Comment by thofma** [comment]
I just pushed it here.

File: Project.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name = "Singular"
 uuid = "bcd08a7b-43d2-5ff7-b6d4-c458787f915c"
-version = "0.21.3"
+version = "0.21.4"
```
Comment (lgoettgens) [inline Project.toml:3]:
```suggestion
version = "0.22. 0"
```

File: Project.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name = "Singular"
 uuid = "bcd08a7b-43d2-5ff7-b6d4-c458787f915c"
-version = "0.21.3"
+version = "0.22. 0"
```
Comment (lgoettgens) [inline Project.toml:3]:
```suggestion
version = "0.22.0"
```

**Comment by thofma** [comment]
Why are we concerned with backporting this time?

**Comment by lgoettgens** [comment]
Once this is merged, there will be no bug fixes delivered to oscar until oscar is adapted to the new Nemo (which may need some time).
But yeah, that's negligible

**Comment by fingolfin** [comment]
We can always make an "emergency" bugfix release from a branch

**Pull request merged**